### PR TITLE
Adds a new datagen input adapter for tables.

### PR DESCRIPTION
**Pull request by gz**
Also reverts chrono to an officially released version.

- [x] Need to add better support for generating strings maybe with this thing https://github.com/cksac/fake-rs
- [x] fix ijson https://github.com/abhizer/ijson/pull/2
- [ ] Support multiple workers (currently limited to 1)
- [ ] Support for inserting batches instead of one row per iteration
- [ ] Use foreign key information 
- [ ] This breaks the WebConsole connectors and Pipeline graph viewer @Karakatiza666 can you have a look at this one? To reproduce follow added tutorial in docs (Part 4)

Is this a user-visible change (yes/no): yes

### CODE REVIEW

File: crates/pipeline-types/src/transport/datagen.rs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+fn default_workers() -> usize {
+    1
+}
+
+fn default_sequence() -> Vec<GenerationPlan> {
+    vec![GenerationPlan::default()]
+}
+
+/// Describe the strategy used to generate values.
```
Comment (ryzhyk) [inline crates/pipeline-types/src/transport/datagen.rs:24]:
```suggestion
/// Strategy used to generate values.
```

File: crates/pipeline-types/src/transport/datagen.rs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize, ToSchema)]
+#[serde(tag = "name")]
+pub enum Strategy {
```
Comment (ryzhyk) [inline crates/pipeline-types/src/transport/datagen.rs:27]: If we plan to expose this type via OpenAPI, it should probably have a qualified name like `DatagenStrategy`.

File: crates/pipeline-types/src/transport/datagen.rs [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    },
+    /// A uniform random distribution is chosen to generate the value.
+    #[serde(rename = "uniform")]
+    Uniform,
+    /// A Zipf distribution is chosen with the specified exponent (`s`).
+    ///
+    /// Note that the Zipf distribution is only available for numbers or types that
+    /// specify `values` or `range`.
+    ///
+    /// - In case `values` is set, the `n` is set to `values.len()`.
```
Comment (ryzhyk) [inline crates/pipeline-types/src/transport/datagen.rs:46]: what's `n`?

File: docs/tutorials/basics/part3.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Part 3: Input and Output Connectors
 
-This is the final part of the tutorial where we will
+This is the key part of the tutorial where we will
```
Comment (ryzhyk) [inline docs/tutorials/basics/part3.md:3]:
```suggestion
In this part of the tutorial where we will
```

File: docs/tutorials/basics/part4.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+-H 'Content-Type: application/json' \
+-H 'Accept: application/json'
+```
+
+You can go and inspect the pipeline and corresponding data in the WebConsole. You see a throughput of 1 record 
+per second with one row in prices changing every second.
+
+To summarize Part 4 of the tutorial,
+
+- We can attach a random generator to Feldera tables to simulate different scenarios such as back-fill, continuous
```
Comment (ryzhyk) [inline docs/tutorials/basics/part4.md:232]:
```suggestion
- We can attach a random generator to Feldera tables to simulate different scenarios such as backfill, continuous 
```

File: docs/tutorials/basics/part4.md [hunk 6]
```diff
@@ -0,0 +1,236 @@
+# Part 4: Random Data Generation
```
Comment (ryzhyk) [inline docs/tutorials/basics/part4.md:1]: link to `datagen` connector docs somewhere in this page.

**Review by ryzhyk — APPROVED** [review]
nice!

File: crates/adapters/src/test/mod.rs [hunk 7]
```diff
@@ -119,18 +121,34 @@ where
...
     config: InputEndpointConfig,
+    relation: Relation,
 ) -> AnyResult<(Box<dyn InputReader>, MockInputConsumer, MockDeZSet<T, U>)>
 where
     T: for<'de> DeserializeWithContext<'de, SqlSerdeConfig> + Send + 'static,
     U: for<'de> DeserializeWithContext<'de, SqlSerdeConfig> + Send + 'static,
 {
-    let (consumer, input_handle) =
-        mock_parser_pipeline(config.connector_config.format.as_ref().unwrap())?;
+    let default_format = FormatConfig {
```
[the remark this answers is not in the document]
Comment (gz) [reply crates/adapters/src/test/mod.rs:130]: It can't because of the `Cow` type I think

**Pull request merged**